src/Components/WebAssembly/Server/src/ContentEncodingNegotiator.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Server
+{
+    internal class ContentEncodingNegotiator
+    {
+        // List of encodings by preference order with their associated extension so that we can easily handle "*".
+        private static readonly StringSegment [] _preferredEncodings =
+            new StringSegment [] { "br", "gzip" };
+
+        private static readonly IDictionary<StringSegment, string> _encodingExtensionMap = new Dictionary<StringSegment, string>(StringSegmentComparer.OrdinalIgnoreCase)
```

Comment:
Just another nitpick:

```suggestion
        private static readonly Dictionary<StringSegment, string> _encodingExtensionMap = new Dictionary<StringSegment, string>(StringSegmentComparer.Ordinal)
```

No need for the interface indirection, and we know the extensions are lowercase because we put them there.

Reply:
These things are ordinalignorecase. See https://github.com/dotnet/aspnetcore/blob/master/src/Middleware/ResponseCompression/src/ResponseCompressionBody.cs#L247